nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -159,6 +159,24 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
 
         self.get_attention_mask_from_fusion = self.cfg.get('get_attention_mask_from_fusion', True)
 
+        # Save activations checkpointing and sequence parallelism parameters to be able to restore them later.
+        self.original_checkpointing_granularity = self.model.language_model.encoder.activations_checkpoint_granularity
+        self.original_checkpointing_num_layers = self.model.language_model.encoder.activations_checkpoint_num_layers
+        self.original_checkpointing_method = self.model.language_model.encoder.activations_checkpoint_method
+        self.original_activations_checkpoint_layers_per_pipeline = (
+            self.model.language_model.encoder.activations_checkpoint_layers_per_pipeline
+        )
+        self.original_sequence_parallel = self.model.language_model.encoder.sequence_parallel
```

Comment:
I don't know that `original` is really needed?

Can we remove this and then, everytime you call `_reset` it saves the current in `last_*` then when you call `_restore` it just restores the `last_*`

Reply:
Sure.

Reply:
Fixed.